tests/Unit/Product/ProductMetaHandlerTest.php

```diff
@@ -0,0 +1,212 @@
+
+		$product_meta_handler->expects( $this->once() )
+							 ->method( 'get' )
+							 ->with( $this->equalTo( $product ), $this->equalTo( $key ) );
+		$product_meta_handler->get_synced_at( $product );
+
+		$product_meta_handler->expects( $this->once() )
+							 ->method( 'delete' )
+							 ->with( $this->equalTo( $product ), $this->equalTo( $key ) );
+		$product_meta_handler->delete_synced_at( $product );
+	}
+
+	public function test_update_type_cast() {
+		$product = WC_Helper_Product::create_simple_product();
+		$this->product_meta_handler->update( $product, ProductMetaHandler::KEY_SYNCED_AT, '12345' );
+		$value = $product->get_meta( '_wc_gla_synced_at', true );
+		$this->assertEquals( 12345, $value );
+	}
+
+	public function test_update() {
+		$product = WC_Helper_Product::create_simple_product();
+		$this->product_meta_handler->update( $product, ProductMetaHandler::KEY_SYNCED_AT, 12345 );
+		$value = $product->get_meta( '_wc_gla_synced_at', true );
+		$this->assertEquals( 12345, $value );
+	}
+
+	public function test_update_throws_exception_invalid_meta_key() {
+		$this->expectException( InvalidMeta::class);
+		$this->product_meta_handler->update( $this->generate_simple_product_mock(), 'invalid_meta_key_test', 1 );
+	}
+
+	public function test_delete() {
+		$product = WC_Helper_Product::create_simple_product();
+		$product->update_meta_data( '_wc_gla_synced_at', 12345 );
+		$product->save_meta_data();
+
+		$this->product_meta_handler->delete( $product, ProductMetaHandler::KEY_SYNCED_AT );
+
+		$value = $product->get_meta( '_wc_gla_synced_at', true );
+		$this->assertEmpty( $value );
+	}
+
+	public function test_delete_throws_exception_invalid_meta_key() {
+		$this->expectException( InvalidMeta::class);
+		$this->product_meta_handler->delete( $this->generate_simple_product_mock(), 'invalid_meta_key_test' );
+	}
+
+	public function test_get_returns_null_if_no_value_exist() {
+		$product = WC_Helper_Product::create_simple_product();
+
+		$value = $this->product_meta_handler->get( $product, ProductMetaHandler::KEY_SYNCED_AT );
+		$this->assertNull( $value );
+	}
+
+	public function test_get() {
+		$product = WC_Helper_Product::create_simple_product();
+		$product->update_meta_data('_wc_gla_synced_at', 12345);
+		$product->save_meta_data();
+
+		$value = $this->product_meta_handler->get( $product, ProductMetaHandler::KEY_SYNCED_AT );
+		$this->assertEquals( 12345, $value );
+	}
+
+	public function test_get_throws_exception_invalid_meta_key() {
+		$this->expectException( InvalidMeta::class);
+		$this->product_meta_handler->get( $this->generate_simple_product_mock(), 'invalid_meta_key_test' );
+	}
+
+	public function test_is_meta_key_valid() {
+		$this->assertFalse( ProductMetaHandler::is_meta_key_valid( 'invalid_meta_key_test' ) );
+		$this->assertTrue( ProductMetaHandler::is_meta_key_valid( ProductMetaHandler::KEY_SYNCED_AT ) );
+	}
+
+	public function test_prefix_meta_query_keys_return_if_not_array() {
```

Comment:
This test name seems confusing to me. I'd guess we were testing un-prefixed keys if they aren't supported.

Reply:
Yeah, this is meant to test the prefix functionality and also whether it returns the unsupported keys intact. I've renamed it and also split the test into two: https://github.com/woocommerce/google-listings-and-ads/pull/854/commits/5e058e994f467290f41a223945811f2b34564262